paimon-core/src/main/java/org/apache/paimon/table/system/SnapshotsTable.java

```diff
@@ -194,11 +195,18 @@ public RecordReader<InternalRow> createReader(Split split) throws IOException {
             Path location = ((SnapshotsSplit) split).location;
             Iterator<Snapshot> snapshots = new SnapshotManager(fileIO, location).snapshots();
             Iterator<InternalRow> rows = Iterators.transform(snapshots, this::toRow);
-            if (projection != null) {
+
+            if (projection != null && projection.length != 0) {
                 rows =
                         Iterators.transform(
                                 rows, row -> ProjectedRow.from(projection).replaceRow(row));
             }
+
+            rows =
+                    Arrays.stream(Iterators.toArray(rows, InternalRow.class))
+                            .sorted(Comparator.comparingLong(row -> row.getLong(0)))
+                            .iterator();
+
```

Comment:
Is it better to proactively add an order statement to the query sql?

Reply:
hi, it will be more complex if we do that, so let's keep it.